packages/ui-extensions-react/src/surfaces/point-of-sale/render.tsx

```diff
@@ -39,22 +39,21 @@ export function reactExtension<Target extends RenderExtensionTarget>(
   return extension<'pos.home.tile.render'>(target as any, async (root, api) => {
     const element = await render(api as ApiForRenderExtension<Target>);
 
-    await new Promise<void>((resolve, reject) => {
+    return new Promise<() => void>((resolve, reject) => {
       try {
-        remoteRender(
+        const remoteRoot = createRoot(root);
+        remoteRoot.render(
           <ExtensionApiContext.Provider value={api}>
             <ErrorBoundary>{element}</ErrorBoundary>
           </ExtensionApiContext.Provider>,
-          root,
-          () => {
-            resolve();
-          },
         );
+
+        return resolve(remoteRoot.unmount);
       } catch (error) {
         // Workaround for https://github.com/Shopify/ui-extensions/issues/325
         // eslint-disable-next-line no-console
         console.error(error);
-        reject(error);
+        return reject(error);
```

Comment:
Not sure what the benefits of this are currently. It would be great to understand the type of error that can be caught here so that we can test to see what benefits this unlocks

Reply:
Not sure that we need a return at all